parser.y

```diff
@@ -6204,12 +6205,21 @@ ShowStmt:
 		// See https://dev.mysql.com/doc/refman/5.7/en/show-grants.html
 		$$ = &ast.ShowStmt{Tp: ast.ShowGrants}
 	}
-|	"SHOW" "GRANTS" "FOR" Username
+|	"SHOW" "GRANTS" "FOR" Username UsingRoles
 	{
 		// See https://dev.mysql.com/doc/refman/5.7/en/show-grants.html
-		$$ = &ast.ShowStmt{
-			Tp:	ast.ShowGrants,
-			User:	$4.(*auth.UserIdentity),
+		if $5 != nil {
+			$$ = &ast.ShowStmt{
+            	Tp:	ast.ShowGrants,
+            	User:	$4.(*auth.UserIdentity),
+            	Roles:  $5.([]*auth.RoleIdentity),
+            }
```

Comment:
Indent lines 6213–6216 with tabs instead of spaces.

Reply:
@kennytm fixed.